### PR TITLE
TEST-#4426: Stop passing is_default kwarg to Modin and pandas.

### LINKED ISSUE #4426: Many groupby tests pass invalid is_default kwarg to pandas and modin.
### Describe the problem
Many groupby tests are passing invalid is_default kwarg to pandas and modin, e.g. [here](https://github.com/modin-project/modin/blob/e5e9634357e60925a5a70e56a1d4882d269f533a/modin/pandas/test/test_groupby.py#L121). In these cases, instead of testing the functions we want to test, we are testing that neither pandas and modin expects the `is_default` argument.

### PR DESCRIPTION
**Pull request by mvashishtha**
Signed-off-by: mvashishtha <mahesh@ponder.io>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4426 
- [N/A] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

**Review by YarShev — APPROVED** [review]
@mvashishtha, good catch, LGTM!

File: modin/pandas/test/test_groupby.py [hunk 1]
```diff
@@ -376,12 +398,26 @@ def maybe_get_columns(df, by):
...
+    )
+    eval_general(
+        modin_groupby,
+        pandas_groupby,
+        lambda df: df.bfill(),
+    )
+    eval_general(
+        modin_groupby,
+        pandas_groupby,
+        lambda df: df.idxmin(),
```
Comment (anmyachev) [inline modin/pandas/test/test_groupby.py:419]: Looks like you've made optional formatting changes. Maybe you are using outdated black?

Please roll them back.

**Review by anmyachev — CHANGES_REQUESTED** [review]

File: modin/pandas/test/test_groupby.py (hunk 1 above)
Comment (mvashishtha) [reply modin/pandas/test/test_groupby.py:419]: Screen Shot 2022-04-27 at 9 28 50 AM

The latest version of black, 22.3.0, is requiring me to reformat the lines this way when I remove `is_default`. Here's a screenshot.

Comment (anmyachev) [reply modin/pandas/test/test_groupby.py:419]: The problem is in the last comma, you don't need to put it.
Example of a more compact representation:
```bash
(modin) C:\Users\amyachev\OneDrive - Intel Corporation\Desktop\projects\modin>black --diff modin 
--- modin\pandas\test\test_groupby.py   2022-04-27 18:44:47.055066 +0000
+++ modin\pandas\test\test_groupby.py   2022-04-27 18:44:55.046452 +0000
@@ -115,13 +115,11 @@
             modin_groupby = modin_df.groupby(by=by[0], as_index=as_index)
             pandas_groupby = pandas_df.groupby(by=by[-1], as_index=as_index)

         modin_groupby_equals_pandas(modin_groupby, pandas_groupby)
         eval_ngroups(modin_groupby, pandas_groupby)
-        eval_general(
-            modin_groupby, pandas_groupby, lambda df: df.ffill()
-        )
+        eval_general(modin_groupby, pandas_groupby, lambda df: df.ffill())
         eval_general(
             modin_groupby,
             pandas_groupby,
             lambda df: df.sem(),
             modin_df_almost_equals_pandas,
would reformat modin\pandas\test\test_groupby.py

All done! ✨ 🍰 ✨
```

Comment (anmyachev) [reply modin/pandas/test/test_groupby.py:419]: The problem is in the last comma, you don't need to put it.
Example of a more compact representation:
```bash
(modin) C:\Users\amyachev\OneDrive - Intel Corporation\Desktop\projects\modin>black --diff modin 
--- modin\pandas\test\test_groupby.py   2022-04-27 18:44:47.055066 +0000
+++ modin\pandas\test\test_groupby.py   2022-04-27 18:44:55.046452 +0000
@@ -115,13 +115,11 @@
             modin_groupby = modin_df.groupby(by=by[0], as_index=as_index)
             pandas_groupby = pandas_df.groupby(by=by[-1], as_index=as_index)

         modin_groupby_equals_pandas(modin_groupby, pandas_groupby)
         eval_ngroups(modin_groupby, pandas_groupby)
-        eval_general(
-            modin_groupby, pandas_groupby, lambda df: df.ffill()
-        )
+        eval_general(modin_groupby, pandas_groupby, lambda df: df.ffill())
         eval_general(
             modin_groupby,
             pandas_groupby,
             lambda df: df.sem(),
             modin_df_almost_equals_pandas,
would reformat modin\pandas\test\test_groupby.py

All done! ✨ 🍰 ✨
```

**Review by anmyachev — APPROVED** [review]
@mvashishtha thanks! LGTM

**Review by YarShev — APPROVED** [review]

**Pull request merged**